synapse/common.py

```diff
@@ -109,6 +109,19 @@ def buid(valu=None):
     byts = s_msgpack.en(valu)
     return hashlib.sha256(byts).digest()
 
+def flatten(item):
```

Comment:
We have a variant of this in synapse/lib/tests/utils.py ( called norm() ). We could replace norm with using this, since it is stricter.

Reply:
This is also very close to hashitem.normitem

Reply:
hashitem calculates the guid of the input value ( and changes dict to tuple ) so i think they are probably different enough to warrant maintaining separation in the interest of not changing them.